sql/catalyst/src/test/scala/org/apache/spark/sql/util/TimestampFormatterSuite.scala

```diff
@@ -83,9 +85,9 @@ class TimestampFormatterSuite extends SparkFunSuite with SQLHelper with Matchers
         2177456523456789L,
         11858049903010203L).foreach { micros =>
         DateTimeTestUtils.outstandingZoneIds.foreach { zoneId =>
-          val formatter = TimestampFormatter(pattern, zoneId)
-          val timestamp = formatter.format(micros)
-          val parsed = formatter.parse(timestamp)
+          val timestamp = TimestampFormatter(pattern, zoneId).format(micros)
+          val parsed = TimestampFormatter(
+            pattern, zoneId, needVarLengthSecondFraction = true).parse(timestamp)
```

Comment:
ditto

Reply:
this change is needed because it is how the actual roundtrip goes no matter what the input is

Reply:
ok makes sense.